numpy/core/arrayprint.py

```diff
@@ -706,6 +706,27 @@ def _extendLine(s, line, word, line_width, next_line_prefix, legacy):
     return s, line
 
 
+def _extendLine_pretty(s, line, word, line_width, next_line_prefix, legacy):
+    """
+    Extends line with nicely formatted (possibly multi-line) string ``word``.
+    """
+    if legacy == '1.13':
+        return _extendLine(s, line, word, line_width, next_line_prefix, legacy)
+
+    words = word.splitlines()
+    line_length = len(line)
+    s, line = _extendLine(
+        s, line, words[0], line_width, next_line_prefix, legacy)
+    for word in words[1::]:
+        s += line.rstrip() + '\n'
+        if line_length + len(word) > line_width \
+                and line_length > len(next_line_prefix):
+            line = next_line_prefix + word
+        else:
+            line = line_length*' ' + word
```

Comment:
This logic looks a little off to me - my reading is that it will do the wrong thing for an object with a repr like
```
short
long enough to cause the line to wrap
short
```
Could you add a test for that?

Reply:
I suspect you actually care about `max(len(word) for word in words))`

Reply:
I supposed this:
```
[None Line 1
 LoooooooooooooooooongLine 2]
```
would be nicer than this (*):
```
[None Line 1
      LoooooooooooooooooongLin
e 2]
```
(last 'endline' due to end of console). However it doesn't always protect us against this ugly 'endline', so it's indeed nonsense. Using `max(len(word) for word in words))` is by far better option:
```
[None 
 Line 1
 LoooooooooooooooooongLine 2]
```
Anyway after some consideration, I would just remove that if clause, so the result would be:
```
[None Line 1
      LoooooooooooooooooongLin
e 2]
```

Reply:
This method is entirely crappy (I hope due to Easter lag). I've changed it a bit